### PR TITLE
[4715] add style to postmaster items that cannot be pulled

### PR DESCRIPTION
**Pull request by goomba5**
Any items that cannot be pulled from Postmaster will be dimmed.
image

### CODE REVIEW

File: src/app/item-popup/ItemPopupBody.tsx [hunk 1]
```diff
@@ -26,7 +26,12 @@ export default function ItemPopupBody({
   onTabChanged(tab: ItemPopupTab): void;
 }) {
   const failureStrings = Array.from(extraInfo?.failureStrings || []);
-  if (!item.canPullFromPostmaster && item.location.inPostmaster) {
+  const engramBucketHash = 375726501;
+  const isNotTransferable =
+    (item.notransfer && !item.isEngram) || (item.notransfer && !item.canPullFromPostmaster);
+  const isEngramCollected = item.location.hash === engramBucketHash;
```
Comment (delphiactual) [inline src/app/item-popup/ItemPopupBody.tsx:32]: Add the following import at the top.
```suggestion
  import { BucketHashes } from 'data/d2/generated-enums';
  const isEngramCollected = item.location.hash === BucketHashes.Engrams;
```

File: src/app/item-popup/ItemPopupBody.tsx [hunk 2]
```diff
@@ -26,7 +27,11 @@ export default function ItemPopupBody({
   onTabChanged(tab: ItemPopupTab): void;
 }) {
   const failureStrings = Array.from(extraInfo?.failureStrings || []);
-  if (!item.canPullFromPostmaster && item.location.inPostmaster) {
+  const isNotTransferable =
+    (item.notransfer && !item.isEngram) || (item.notransfer && !item.canPullFromPostmaster);
```
Comment (delphiactual) [inline src/app/item-popup/ItemPopupBody.tsx:31]:
```suggestion
        (item.notransfer && (!item.isEngram || !item.canPullFromPostmaster);
```

File: src/app/inventory/DraggableInventoryItem.tsx [hunk 3]
```diff
@@ -110,6 +110,9 @@ function DraggableInventoryItem({
   const [touchActive, setTouchActive] = useState(false);
   const longPressed = useRef<boolean>(false);
   const timer = useRef<number>(0);
+  const isNotTransferable =
+    (item.notransfer && !item.isEngram) || (item.notransfer && !item.canPullFromPostmaster);
```
Comment (delphiactual) [inline src/app/inventory/DraggableInventoryItem.tsx:114]:
```suggestion
    (item.notransfer && (!item.isEngram || !item.canPullFromPostmaster);
```

**Review by delphiactual — APPROVED** [review]

**Comment by bhollis** [comment]
Thanks! Would you mind adding a screenshot to the PR description?

**Comment by goomba5** [comment]
@bhollis more than welcome! I'll add a screenshot real quick. Thank you @delphiactual for reviewing my code and not torching me.

**Review by bhollis — APPROVED** [review]

**Comment by nev-r** [comment]
@goomba5 any interest in taking a look at #6473?

**Comment by goomba5** [comment]
@nev-r absolutely. I think I know why it's giving the incorrect message.

**Pull request merged**